src/ros_tcp_endpoint/server.py

```diff
@@ -51,7 +58,9 @@ def __init__(self, node_name, buffer_size=1024, connections=10):
         self.connections = connections
         self.syscommands = SysCommands(self)
 
-    def start(self):
+    def start(self, source_destination_dict=None):
+        if source_destination_dict is not None:
+            self.source_destination_dict = source_destination_dict
```

Comment:
What happens downstream if source_destination_dict is None?  Is that supported behavior?

Reply:
Yes, that's fine. Most likely, self.source_destination_dict will remain set to the value from the constructor, namely `{}`; then you can add handlers by calling RegisterSubscriber etc.

Reply:
To be clear, the intent of this argument is to provide a natural place to set source_destination_dict. It can't be done in the constructor because many handlers expect a reference to the server. So passing it into start seems more natural than the current solution, which is just to set it from outside. See: https://github.com/Unity-Technologies/Unity-Robotics-Hub/blob/main/tutorials/ros_packages/robotics_demo/scripts/server_endpoint.py